### DISCUSSION TITLE
Is there any solution

### DISCUSSION THREAD

**Discussion by eminozdgnn1** [chakra-ui/chakra-ui#8766] [General] [answered]
Let me try to explain what's happening here:

With our current setup and approach, during SSR, emotion will render `style` tags inline as a sibling of any Chakra component. After hydration, emotion will move the styles to the head as fast as it can to prevent FOUC (flash of unstyled content)

If we implemented the SSR setup recommended by emotion docs, we'll skip that hydration step and extract all the styles to the head ahead of time. I think we should attempt to do this within our docs @nikolovlazar @TylerAPfledderer. If we notice improvements from this, we'll update our documentation to reflect this.

When you use markup validators, they'll run against the SSR output of the website and will see inline `style` as invalid, when in fact, that's not the full story.

_Originally posted by @segunadebayo in https://github.com/chakra-ui/chakra-ui-docs/issues/644#issuecomment-1146753917_

**Comment by segunadebayo** [comment, marked as answer]
Closing. This is an old SSR note, not an open question.
